schema/pgsql/schema.sql

```diff
@@ -0,0 +1,1904 @@
+COMMENT ON COLUMN hostgroup_member.environment_id IS 'environment.id';
+COMMENT ON COLUMN hostgroup_member.host_id IS 'host.id';
+COMMENT ON COLUMN hostgroup_member.hostgroup_id IS 'hostgroup.id';
+
+CREATE TABLE host_customvar (
+  id bytea20 NOT NULL,
+  environment_id bytea20 NOT NULL,
+  host_id bytea20 NOT NULL,
+  customvar_id bytea20 NOT NULL,
+
+  CONSTRAINT pk_host_customvar PRIMARY KEY (id)
+);
+
+ALTER TABLE host_customvar ALTER COLUMN id SET STORAGE PLAIN;
+ALTER TABLE host_customvar ALTER COLUMN environment_id SET STORAGE PLAIN;
+ALTER TABLE host_customvar ALTER COLUMN host_id SET STORAGE PLAIN;
+ALTER TABLE host_customvar ALTER COLUMN customvar_id SET STORAGE PLAIN;
+
+CREATE INDEX idx_host_customvar_host_id ON host_customvar(host_id, customvar_id);
+CREATE INDEX idx_host_customvar_customvar_id ON host_customvar(customvar_id, host_id);
+
+COMMENT ON COLUMN host_customvar.id IS 'sha1(environment.id + host_id + customvar_id)';
+COMMENT ON COLUMN host_customvar.environment_id IS 'environment.id';
+COMMENT ON COLUMN host_customvar.host_id IS 'host.id';
+COMMENT ON COLUMN host_customvar.customvar_id IS 'customvar.id';
+
+CREATE TABLE hostgroup_customvar (
+  id bytea20 NOT NULL,
+  environment_id bytea20 NOT NULL,
+  hostgroup_id bytea20 NOT NULL,
+  customvar_id bytea20 NOT NULL,
+
+  CONSTRAINT pk_hostgroup_customvar PRIMARY KEY (id)
+);
+
+ALTER TABLE hostgroup_customvar ALTER COLUMN id SET STORAGE PLAIN;
+ALTER TABLE hostgroup_customvar ALTER COLUMN environment_id SET STORAGE PLAIN;
+ALTER TABLE hostgroup_customvar ALTER COLUMN hostgroup_id SET STORAGE PLAIN;
+ALTER TABLE hostgroup_customvar ALTER COLUMN customvar_id SET STORAGE PLAIN;
+
+CREATE INDEX idx_hostgroup_customvar_hostgroup_id ON hostgroup_customvar(hostgroup_id, customvar_id);
+CREATE INDEX idx_hostgroup_customvar_customvar_id ON hostgroup_customvar(customvar_id, hostgroup_id);
+
+COMMENT ON COLUMN hostgroup_customvar.id IS 'sha1(environment.id + hostgroup_id + customvar_id)';
+COMMENT ON COLUMN hostgroup_customvar.environment_id IS 'environment.id';
+COMMENT ON COLUMN hostgroup_customvar.hostgroup_id IS 'hostgroup.id';
+COMMENT ON COLUMN hostgroup_customvar.customvar_id IS 'customvar.id';
+
+CREATE TABLE host_state (
+  id bytea20 NOT NULL,
+  host_id bytea20 NOT NULL,
+  environment_id bytea20 NOT NULL,
+  properties_checksum bytea20 NOT NULL,
+
+  state_type state_type NOT NULL,
+  soft_state tinyuint NOT NULL,
+  hard_state tinyuint NOT NULL,
+  previous_hard_state tinyuint NOT NULL,
+  attempt tinyuint NOT NULL,
+  severity smalluint NOT NULL,
+
+  output text DEFAULT NULL,
+  long_output text DEFAULT NULL,
+  performance_data text DEFAULT NULL,
+  normalized_performance_data text DEFAULT NULL,
+
+  check_commandline text DEFAULT NULL,
+
+  is_problem boolenum NOT NULL,
+  is_handled boolenum NOT NULL,
+  is_reachable boolenum NOT NULL,
+  is_flapping boolenum NOT NULL,
+  is_overdue boolenum NOT NULL DEFAULT 'n',
```

Comment:
The state sync would error.

Reply:
Otherwise you explicitly have to give a value and this happens to be the only one where we don't do this? So if we do add boolean columns in the future, we might have to add them with `DEFAULT 'n'` as well to keep the schema compatible.

If we keep the `boolenum` ([which I'm still not a fan of to be honest](https://github.com/Icinga/icingadb/pull/221#discussion_r755860868)), maybe add the default to the type definition instead to keep all these columns consistent in the long-term?

Reply:
CC @lippserd

Reply:
The `DEFAULT 'n'` should be part of the `CREATE TYPE` specification as we mirror the behavior of MySQL.

Reply:
Oh noes!

Reply:
I thought I explicitly checked this in the docs, but looks like that was for domains and not types: https://www.postgresql.org/docs/14/sql-createdomain.html

Reply:
I see. Is `CREATE DOMAIN boolenum AS CHAR(1) CHECK (VALUE IN ('n', 'y')) DEFAULT 'n';` an option? I did not check how to force a specific sort order.

Reply:
Sort order.. ah! Right! n < y < sticky e.g.!